vtex-apps/address-locator
# Identify user with phone number

#### What is the purpose of this pull request?
This PR only pertains to the second tab of the screen, "Redeem Address". The first tab is on #2 .
Created the view with input and validation of brazilian phone numbers, using components from '@vtex/profile-form'. It's only Brazil for now but the code is prepared to add other countries in the future.
After validation, the number is used to query for the user email, and if that exists, sets it on the orderForm. The redirection after the user is identified will be done when this is integrated with `vtex.delivery-dreamstore`.

#### What problem is this solving?
Input, formatting and validation of phone numbers, specific to Brazil.
Getting user profile from the provided phone number

#### How should this be manually tested?
Go to the second tab of the UI (Redeem Address)
Try inputting different phone numbers. if they are not valid, the UI should update showing error messages.
When submitting the form with any number, check if it shows an error message that the profile was not found.
I created one profile with the number 21987654321, try inputting that and seeing the network tab activity, it should update the orderform. You can query the orderForm afterwards and see that the email is now on `clientProfileData`. If you want to again after this step, you need to delete the cookie `checkout.vtex.com` and reload the page to get a new orderform

#### Screenshots or example usage
(Workspace)[https://addressredeempr--delivery.myvtex.com/]

#### Types of changes
